device/cuda/src/seeding/seed_finding.cu

```diff
@@ -162,28 +172,42 @@ vecmem::data::vector_buffer<seed> seed_finding::operator()(
     const sp_grid_const_view& g2_view,
     const std::vector<unsigned int>& grid_sizes) const {
 
-    // Get the prefix sum for the spacepoint grid using buffer.
-    const device::prefix_sum_t sp_grid_prefix_sum = device::get_prefix_sum(
-        grid_sizes, (m_mr.host ? *(m_mr.host) : m_mr.main));
+    // Create prefix sum of grid_sizes vector
+    std::vector<unsigned int> grid_sizes_sum(grid_sizes.size());
+    std::partial_sum(grid_sizes.begin(), grid_sizes.end(),
+                     grid_sizes_sum.begin(), std::plus<unsigned int>());
+    const unsigned int& totalGrid = grid_sizes_sum[grid_sizes_sum.size() - 1];
 
-    // Set up the doublet counter buffer.
-    device::doublet_counter_container_types::buffer doublet_counter_buffer =
-        device::make_doublet_counter_buffer(grid_sizes, *m_copy, m_mr.main,
-                                            m_mr.host);
+    // Create buffer and view objects for summation of sizes
+    vecmem::data::vector_buffer<unsigned int> grid_sizes_buff(
+        grid_sizes_sum.size(), m_mr.main);
+    m_copy->setup(grid_sizes_buff);
+    (*m_copy)(vecmem::get_data(grid_sizes_sum), grid_sizes_buff);
+    vecmem::data::vector_view<unsigned int> grid_sizes_view(grid_sizes_buff);
```

Comment:
This whole copy should be dependent on whether we use 2 memory resources, or just one. (I still fear that we don't do this correctly in many parts of the code.)

If we use just one memory resource, then the kernel should receive `vecmem::get_data(grid_sizes_sum)` directly.